src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -315,31 +437,94 @@ struct FabricSceneData : public PersistentData<kPersistentFabricBufferMax>
             // If Find doesn't return CHIP_NO_ERROR, the scene wasn't found, which doesn't return an error
             VerifyOrReturnValue(this->Find(scene_id, scene.index) == CHIP_NO_ERROR, CHIP_NO_ERROR);
 
+            // Update the global scene count
+            GlobalSceneCount global_scene_count;
+            ReturnErrorOnFailure(global_scene_count.Load(storage));
+            global_scene_count.count_value--;
+            ReturnErrorOnFailure(global_scene_count.Save(storage));
+
             scene_count--;
             scene_map[scene.index].Clear();
-            ReturnErrorOnFailure(this->Save(storage));
+            err = this->Save(storage);
+
+            if (err != CHIP_NO_ERROR)
+            {
+                global_scene_count.count_value++;
+                ReturnErrorOnFailure(global_scene_count.Save(storage));
+            }
 
             err = scene.Delete(storage);
 
             // On failure to delete scene, undoes the change to the Fabric Scene Data
             if (err != CHIP_NO_ERROR)
             {
+                global_scene_count.count_value++;
+                ReturnErrorOnFailure(global_scene_count.Save(storage));
+
                 scene_count++;
                 scene_map[scene.index] = scene.mStorageId;
                 ReturnErrorOnFailure(this->Save(storage));
             }
         }
         return err;
     }
+
+    CHIP_ERROR Load(PersistentStorageDelegate * storage) override
+    {
+        VerifyOrReturnError(nullptr != storage, CHIP_ERROR_INVALID_ARGUMENT);
+
+        uint8_t buffer[kPersistentFabricBufferMax] = { 0 };
+        StorageKeyName key                         = StorageKeyName::Uninitialized();
+
+        // Set data to defaults
+        Clear();
+
+        // Update storage key
+        ReturnErrorOnFailure(UpdateKey(key));
+
+        // Load the serialized data
+        uint16_t size  = static_cast<uint16_t>(sizeof(buffer));
+        CHIP_ERROR err = storage->SyncGetKeyValue(key.KeyName(), buffer, size);
+        VerifyOrReturnError(CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND != err, CHIP_ERROR_NOT_FOUND);
+        ReturnErrorOnFailure(err);
+
+        // Decode serialized data
+        TLV::TLVReader reader;
+        reader.Init(buffer, size);
+
+        err = Deserialize(reader);
+        // If PersistentData::Load returns "buffer too small", the table in flash memory holds to many scenes (Can happend if
```

Comment:
Not clear where is PersistentData::Load is from this comment. I assume you mean the load in `Deserialize`

Reply:
Indeed, Deserialize was suppose to be there instead.

Reply:
Fixed in latest commit.